dist/src/main/java/io/camunda/zeebe/shared/management/ClusterEndpoint.java

```diff
@@ -219,6 +249,85 @@ private static PartitionStateCode mapPartitionState(final State state) {
     };
   }
 
+  private static GetTopologyResponse mapClusterTopology(final ClusterTopology topology) {
+    final var response = new GetTopologyResponse();
+    final List<BrokerState> brokers = mapBrokerStates(topology.members());
+    final TopologyChange change;
+
+    if (topology.pendingChanges().isPresent()) {
+      change = mapOngoingChange(topology.pendingChanges().get());
+    } else if (topology.lastChange().isPresent()) {
+      change = mapCompletedChange(topology.lastChange().get());
+    } else {
+      change = new TopologyChange();
+    }
+
+    response.version(topology.version()).brokers(brokers).change(change);
+
+    return response;
+  }
+
+  private static TopologyChange mapCompletedChange(final CompletedChange completedChange) {
+    return new TopologyChange()
+        .id(completedChange.id())
+        .status(mapChangeStatus(completedChange.status()))
+        .startedAt(completedChange.startedAt().atOffset(ZoneOffset.UTC))
+        .completedAt(completedChange.completedAt().atOffset(ZoneOffset.UTC));
+  }
+
+  private static TopologyChange mapOngoingChange(final ClusterChangePlan clusterChangePlan) {
+    return new TopologyChange()
+        .id(clusterChangePlan.id())
+        .status(mapChangeStatus(clusterChangePlan.status()))
+        .pending(mapOperations(clusterChangePlan.pendingOperations()))
+        .completed(mapCompletedOperations(clusterChangePlan.completedOperations()));
+  }
+
+  private static StatusEnum mapChangeStatus(final Status status) {
+    return switch (status) {
+      case IN_PROGRESS -> StatusEnum.IN_PROGRESS;
+      case COMPLETED -> StatusEnum.COMPLETED;
+      case FAILED -> StatusEnum.FAILED;
+    };
+  }
+
+  private static List<TopologyChangeCompletedInner> mapCompletedOperations(
+      final List<CompletedOperation> completedOperations) {
+    return completedOperations.stream().map(ClusterEndpoint::mapCompletedOperation).toList();
+  }
+
+  private static TopologyChangeCompletedInner mapCompletedOperation(
```

Comment:
:upside_down_face: We cannot reuse `mapOperation` because the generated model has two different types.

Reply:
Oh how annoying :disappointed: 

The only way around it that I see is to make have every `Operation` carry a `startedAt`, `completedAt` timestamp.
It's a decent-size refactoring but would allow us to keep only a simple list of `TopologyChangeOperation` in `ClusterChangePlan`. Operations without a `completedAt` timestamp are pending, otherwise they are completed.

Reply:
I think it is not much refactoring if we only change the spec. So the spec defines on `Operation` that has  `completedAt` timestamp. `completedAt` is only present for completed operations. For pending operations it will be empty/null.

Reply:
Or we change the spec to match the internal representation. `completedOperation = {operation= xxx, completeAt=xxx}`, instead of a flat structure.

Reply:
I think I slightly prefer a flat structure where timestamps are null because it looks a bit better in JSON. But feel free to pick whatever you prefer